ouroboros-consensus/test-storage/Test/Ouroboros/Storage/VolatileDB.hs

```diff
@@ -17,63 +17,44 @@ import           Control.Monad.Class.MonadThrow
 import           Test.Tasty (TestTree, testGroup)
 import           Test.Tasty.HUnit
 import           Test.Tasty.QuickCheck (testProperty)
 
-import           Ouroboros.Consensus.Util (SomePair (..))
-
 import           Ouroboros.Storage.FS.API
-import           Ouroboros.Storage.FS.API.Types
-import           Ouroboros.Storage.Util.ErrorHandling (ErrorHandling (..))
+import           Ouroboros.Storage.Util.ErrorHandling (ErrorHandling)
 import qualified Ouroboros.Storage.Util.ErrorHandling as EH
 import           Ouroboros.Storage.VolatileDB.API
-import           Ouroboros.Storage.VolatileDB.Impl (openDB)
+import           Ouroboros.Storage.VolatileDB.Impl
 import qualified Ouroboros.Storage.VolatileDB.Impl as Internal hiding (openDB)
-import           Ouroboros.Storage.VolatileDB.Util (modifyTMVar)
 import           Test.Ouroboros.Storage.Util
 import qualified Test.Ouroboros.Storage.VolatileDB.StateMachine as StateMachine
+import           Test.Ouroboros.Storage.VolatileDB.TestBlock
+
+withTestDB :: (HasCallStack, MonadSTM m, MonadMask m)
+           => HasFS m h
+           -> ErrorHandling (VolatileDBError BlockId) m
+           -> Parser m BlockId
+           -> Int
+           -> (VolatileDB BlockId m -> m a)
+           -> m a
+withTestDB hasFS err parser n = withDB (openDB hasFS err parser n toSlot)
 
 tests :: HasCallStack => TestTree
 tests = testGroup "VolatileDB"
   [
     testProperty "list parsing round trips" prop_roundTrips
-  , testProperty "put/get rounftrips" prop_VolatileRoundTrips
-  , testProperty "reOpen" prop_VolatileReOpenState
-  , testProperty "reOpen and Write" prop_VolatileReOpenWrite
-  , testProperty "garbage collect" prop_VolatileGarbageCollect
-  , testProperty "garbage collect state" prop_VolatileGarbageState
-  , testProperty "remove empty file to Write" prop_VolatileReopenRemoveEmptyFile
-  , testProperty "no slot returns Nothing" prop_VolatileNoSlot
-  , testProperty "duplicated slot" prop_VolatileDuplicatedSlot
-  , testProperty "duplicated slot" prop_VolatileDecodeFail
-  , testProperty "undisputable lookup" prop_VolatileUndisputableLookup
-  , testProperty "undisputable lookup with closed handle" prop_VolatileUndisputableLookupClose
   , testProperty "Invalid argument" prop_VolatileInvalidArg
-  , testProperty "Invalid filename" prop_VolatileParseFileNameError
-  , testProperty "Invalid number of blocks" prop_VolatileSlotsPerFileError
+  , testProperty "garbage collect" prop_VolatileGarbageCollect
   , StateMachine.tests
   ]
 
-withTestDB :: (HasCallStack, MonadSTM m, MonadCatch m)
-           => HasFS m h
-           -> ErrorHandling (VolatileDBError MyBlockId) m
-           -> FsPath
-           -> Parser m MyBlockId
-           -> Int
-           -> (VolatileDB MyBlockId m -> m a)
-           -> m a
-withTestDB hasFS err path parser n = withDB (openDB hasFS err path parser n toSlot)
-
-prop_roundTrips :: [MyBlockId] -> Property
+prop_roundTrips :: HasCallStack => [BlockId] -> Property
```

Comment:
this can't be replaced by q-s-m since it tests only the parser fr the db (not the db itself)